### PR TITLE
chore: update bucketing distribution tests for numbers with 4 decimal places

### PR DESCRIPTION
**Pull request by kaushalkapasi**
- chore: update WASM bucketing test data to include config with 4 decimal places for a random distribution 
- chore: update JS bucketing test to include config with 4 decimal places for a random distribution

### CODE REVIEW

File: lib/shared/bucketing/__tests__/bucketing.test.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ describe('User Hashing and Bucketing', () => {
             _audience: { _id: 'id', filters: [] } as unknown as Audience,
             _id: 'target',
             distribution: [
-                { _variation: 'var1', percentage: 0.25 },
-                { _variation: 'var2', percentage: 0.45 },
+                { _variation: 'var1', percentage: 0.2555 },
```
Comment (jonathannorris) [inline lib/shared/bucketing/__tests__/bucketing.test.ts:28]: you should probably update the expected value ranges below.

Comment (kaushalkapasi) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: updated to narrow the expected range to be 0.5% around `var1` and `var2`

Comment (kaushalkapasi) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: @jonathannorris thoughts on the ranges.. these tests will flake even with 50k evals due to random distributions being, well random 😅

Comment (jonathannorris) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: @kaushalkapasi has that flakeness changed? it shouldn't be that bad...

Comment (kaushalkapasi) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: nope.. with 4 decimal places and the narrowing of the range checks (was previously  +/- 1% of the distribution percentage, now its +/- 0.5%) there's just a higher likely hood of things being just outside that +/- 0.5%

Comment (kaushalkapasi) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: the existing tests cover that the config is saved and parsed correctly.. should we add an explicit check for the 4 decimal places? or is the existing test that the config is valid / matches the input config when parsed out of wasm enough?

Comment (jonathannorris) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: does that test config have these longer decimal places in them?

**Review by jonathannorris — APPROVED** [review]

File: lib/shared/bucketing/__tests__/bucketing.test.ts (hunk 1 above)
Comment (kaushalkapasi) [reply lib/shared/bucketing/__tests__/bucketing.test.ts:28]: yeah, I've updated the underlying `lib/shared/bucketing-test-data/src/data/testData.ts` file to include a couple of instances to use numbers with 4 decimal places

**Pull request merged**